From 05abd666016db971d8738685a31277771d16b29d Mon Sep 17 00:00:00 2001
From: Armin Kuster <akuster808@gmail.com>
Date: Sun, 4 Jul 2021 08:50:06 -0700
Subject: [PATCH] fix musl build failures

Missing _PATH_HOSTS and some NETDB defines when musl is enabled.

These are work arounds for now while we figure out where the real fix should reside (musl, gcompact, sssd):

./sssd-2.5.1/src/providers/fail_over.c:1199:19: error: '_PATH_HOSTS' undeclared (first use in this function)
|  1199 |                   _PATH_HOSTS);
|       |                   ^~~~~~~~~~~

and

i./sssd-2.5.1/src/sss_client/nss_ipnetworks.c:415:21: error: 'NETDB_INTERNAL' undeclared (first use in this function)
|   415 |         *h_errnop = NETDB_INTERNAL;

Upstream-Status: Pending
Signed-off-by: Armin Kuster <akuster808@gmail.com>
---
 src/providers/fail_over.c | 4 ++++
 src/sss_client/sss_cli.h  | 8 ++++++++
 2 files changed, 12 insertions(+)

diff --git a/src/providers/fail_over.c b/src/providers/fail_over.c
index d56c20f..2c6ceb6 100644
--- a/src/providers/fail_over.c
+++ b/src/providers/fail_over.c
@@ -33,6 +33,10 @@
 #include <netdb.h>
 #include <arpa/inet.h>
 
+#if !defined(_PATH_HOSTS)
+#define _PATH_HOSTS     "/etc/hosts"
+#endif
+
 #include "util/dlinklist.h"
 #include "util/refcount.h"
 #include "util/util.h"
diff --git a/src/sss_client/sss_cli.h b/src/sss_client/sss_cli.h
index 57e1ead..f58510f 100644
--- a/src/sss_client/sss_cli.h
+++ b/src/sss_client/sss_cli.h
@@ -52,6 +52,14 @@ typedef int errno_t;
 #define NETDB_INTERNAL (-1)
 #endif
 
+#ifndef NETDB_INTERNAL
+# define NETDB_INTERNAL (-1)
+#endif
+
+#ifndef NETDB_SUCCESS
+# define NETDB_SUCCESS (0)
+#endif
+
 #define SSS_NSS_PROTOCOL_VERSION 1
 #define SSS_PAM_PROTOCOL_VERSION 3
 #define SSS_SUDO_PROTOCOL_VERSION 1
